### PR TITLE
Update tests to support latest ripple-lib:

### PR DESCRIPTION
**Pull request by sublimator**
* Update ripple-lib api usage
* Use latest npm ripple-lib
  * Tested with bignumber.js branch and tip of develop
* Use new version of coffee-script
   * Better source maps
* Update mocha
* Add assert-diff for better error reporting
* Add rconsole, enabled via USE_RCONSOLE env var

@ximinez @wltsmrz

### CODE REVIEW

**Comment by sublimator** [comment]
Hrmm, looks like rconsole doesn't want to build with linux/clang

**Comment by ximinez** [comment]
In Windows:
```
$ npm test

> rippled@0.0.1 test c:\Dev\rippled-merge
> mocha test/websocket-test.js test/server-test.js test/*-test.{js,coffee}



  WebSocket connection
    1) "before each" hook

  0 passing (15ms)
  1 failing

  1) WebSocket connection "before each" hook:
     Uncaught TypeError: Bad argument
    at ChildProcess.spawn (child_process.js:936:24)
    at Object.exports.spawn (child_process.js:736:9)
    at Server._serverSpawnSync (c:\Dev\rippled-merge\test\server.js:111:22)
    at c:\Dev\rippled-merge\test\server.js:172:10
    at c:\Dev\rippled-merge\node_modules\mocha\node_modules\glob\node_modules\gr
aceful-fs\graceful-fs.js:104:5
    at Object.oncomplete (fs.js:107:15)




npm ERR! Test failed.  See above for more details.
npm ERR! not ok code 0
```

**Comment by ximinez** [comment]
*Update* Above comment is a sort-of false alarm. Your branch does not include 0.27.0-b10, which includes the `--rippled` parameter feature for npm tests, which I am using. If I change my config, the tests pass fine in Linux and Windows.

File: test/batmans-belt.coffee [hunk 1]
```diff
@@ -101,6 +101,9 @@ exports.submit_for_final = (tx, done) ->
   accept so the transaction will finalize.
 
   '''
-  tx.on 'proposed', -> tx.remote.ledger_accept()
+  tx.on 'submitted', (m) ->
+    ter = (m.engine_result ? '').slice(0, 3)
+    if ter in ['tes', 'tec']
+      tx.remote.ledger_accept()
```
Comment (ximinez) [inline test/batmans-belt.coffee]: Is `proposed` going away? Or are there tests where we expect a "tec" code, and need that transaction in a closed ledger?

File: test/ledger-state.coffee [hunk 2]
```diff
@@ -357,12 +357,12 @@ exports.LedgerState = class LedgerState
...
 
     base_reserve = @parse_amount base_reserve_amt
-    inc_reserve = @parse_amount incr_reserve_amt
+    incr_reserve = @parse_amount incr_reserve_amt
 
     @assert base_reserve != null,
            "Base reserve amount #{base_reserve_amt} is invalid"
 
-    @assert base_reserve != null,
+    @assert incr_reserve != null,
```
Comment (ximinez) [inline test/ledger-state.coffee]: Good catch!

**Comment by ximinez** [comment]
Assuming the answer to the `proposed` question above is ok, then the changes look good.

:+1:

**Comment by sublimator** [comment]
I'd like Brandon's input re: proposed. I can't recall offhand now but
there's been some changes to that event. I think it's no longer emitted
unless the result is tesSUCCESS.

**Comment by sublimator** [comment]
Sorry, yeah, we expect tec sometimes to ensure things like tecFUNDS_FROZEN
(or whatever it is, on mobile atm) work.

**Comment by sublimator** [comment]
@wltsmrz

**Pull request closed without merging**